src/libraries/Common/src/System/Net/Security/SecurityBuffer.Windows.cs

```diff
@@ -28,6 +29,62 @@ namespace System.Net.Security
         private SecurityBuffer _item2;
     }
 
+    [StructLayout(LayoutKind.Sequential)]
+    internal ref struct InputSecurityBuffers
```

Comment:
It feels like we could use a `ValueSpanBuilder` to match `ValueStringBuilder` :)

Reply:
The problem is you can't stackalloc structs that have references in them.

Reply:
I had one prototype to pass Span<byte> all the way down to low-level PAL to avoid the need for encapsulating type. But that was kind of ugly. 

In practice, we need at most 2 buffers -  input data and optionally either token binding or ALPN. For SSL, second buffer is always EMPTY, following SHOULD recommendation from schannel API. But it seems to work fine without it as well as NegotiateStream works fine if the extra EMPTY buffer is passed in.